src/saber/infos.ts

```diff
@@ -17,13 +17,21 @@ const DIGIT = new BN(10000000000);
 let infos: IInstancePool & IInstanceFarm;
 
 infos = class InstanceSaber {
-  static async getAllPools(connection: Connection): Promise<PoolInfo[]> {
+  static async getAllPools(connection: Connection, page?: PageConfig): Promise<PoolInfo[]> {
     const sizeFilter: DataSizeFilter = {
       dataSize: 395,
     };
     const filters = [sizeFilter];
     const config: GetProgramAccountsConfig = { filters: filters };
-    const poolAccounts = await connection.getProgramAccounts(POOL_PROGRAM_ID, config);
+    let poolAccounts = await connection.getProgramAccounts(POOL_PROGRAM_ID, config);
+    poolAccounts.sort((a, b) => a.pubkey.toBase58().localeCompare(b.pubkey.toBase58()));
+
+    if (page) {
+      let batchSize = poolAccounts.length / page.pageSize;
```

Comment:
1. Perhaps this piece of logic can be extracted to utils as a function? Ex: `Paginate`
2. Assign the batched accounts to a new variable (Ex: `pagedPoolAccounts`) instead of mutating `allV4AMMAccount`